### ISSUE TITLE
How to get models from NNSearch? Should there be an NNSearch abstract class before the Cosine class?

### ISSUE THREAD

**Issue by mithasalsabila** [olha-kaminska/frnn_emotion_detection#1]
- I'm trying to classify data with Doc2Vec and FRNN-OWA methods. For the FRNN-OWA code I tried to implement from your code: frnn_owa_eval.py , but when I try to run the Cosine class I get an error: **type object 'NNSearch' has no attribute 'Model'**
- Then I tried to git clone https://github.com/oulenz/fuzzy-rough-learn.git , and it worked when running the Cosine class. But when I try to run the accuracy command which calls the 'cross_validation_ensemble_owa' function, an error appears: **AttributeError: 'Model' object has no attribute 'X_T' .** It was from the Cosine class.

![image](https://user-images.githubusercontent.com/67451561/124386671-a5baf480-dd05-11eb-9709-290f9e508ed9.png)

**Comment by olha-kaminska** [comment]
The author of the **fuzzy-rough-learn** package is developing the new version, so some changes were done and more to go. It can cause issues for some functions from my repo, like the one that you mentioned. I would recommend using the previous version 0.0.1 of the **fuzzy-rough-learn** package. Once the new version will be released, I will update my code and close this issue.

**Comment by CanerCo** [comment]
I am currently experiencing an issue while attempting to execute the frnn_owa_eval.py script. The error message I encounter is: AttributeError: type object 'NNSearch' has no attribute 'Model'. I have already verified that I am using the latest version of the repository and have installed all the required packages, but the problem persists. Additionally, I attempted to install the package by cloning the repository at https://github.com/oulenz/fuzzy-rough-learn.git (version 0.2.2 as of today), but encountered a different error this time: ModuleNotFoundError: No module named 'frlearn'.

any guidance on how to resolve this problem?
